src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Item>(
+  items: Item[],
+  key: keyof Item,
```

Comment:
Consider using a more specific type constraint for 'key'. Instead of 'keyof Item', you could use 'keyof Item & string' to ensure that the key is actually a string property of Item, since it is being used as a property name in an object.